.github/workflows/pkgci_regression_test_cpu.yml

```diff
@@ -71,14 +71,58 @@ jobs:
               --config-files=build_tools/pkgci/external_test_suite/onnx_cpu_llvm_sync.json
-      # TODO(scotttodd): add a local cache for these large files to a persistent runner
+  # persistent runner with cache enabled for model testing
+  linux_x86_64_models:
+    name: Linux (x86_64) Model Testing/Benchmark
+    runs-on: nodai-amdgpu-w7900-x86-64
+    env:
+      PACKAGE_DOWNLOAD_DIR: ${{ github.workspace }}/.packages
+      IREERS_ARTIFACT_DIR: ${{ github.workspace }}/artifacts
+      VENV_DIR: ${{ github.workspace }}/venv
+      IREE_TEST_FILES: ~/iree_tests_cache
+    steps:
+      - name: Checking out IREE repository
+        uses: actions/checkout@8f4b7f84864484a7bf31766abe9204da3cbe65b3 # v3.5.0
+        with:
+          submodules: false
+      - uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1  # v4.7.0
+        with:
+          # Must match the subset of versions built in pkgci_build_packages.
+          python-version: '3.11'
+      - uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a  # v3.0.2
+        with:
+          name: linux_x86_64_release_packages
+          path: ${{ env.PACKAGE_DOWNLOAD_DIR }}
+      - name: Setup venv
+        run: |
+          ./build_tools/pkgci/setup_venv.py ${VENV_DIR} \
+            --artifact-path=${PACKAGE_DOWNLOAD_DIR} \
+            --fetch-gh-workflow=${{ inputs.artifact_run_id }}
+
+      # Out of tree tests
+      - name: Check out external TestSuite repository
+        uses: actions/checkout@8f4b7f84864484a7bf31766abe9204da3cbe65b3 # v3.5.0
+        with:
+          repository: nod-ai/SHARK-TestSuite
+          ref: 1d77dcffdffc658e4defa6021a60ca1e9d065c61
+          path: SHARK-TestSuite
+          submodules: false
+          lfs: true
+      - name: Install external TestSuite Python requirements
+        run: |
+          source ${VENV_DIR}/bin/activate
+          python -m pip install -r SHARK-TestSuite/iree_tests/requirements.txt
       - name: Download remote files for real weight model tests
         run: |
           source ${VENV_DIR}/bin/activate
-          python SHARK-TestSuite/iree_tests/download_remote_files.py
+          python SHARK-TestSuite/iree_tests/download_remote_files.py --root-dir pytorch/models
       - name: Run external tests - Models with real weights
         run: |
           source ${VENV_DIR}/bin/activate
-          pytest SHARK-TestSuite/iree_tests/pytorch/ -k real_weights \
+          pytest SHARK-TestSuite/iree_tests/pytorch/models -k real_weights \
                -rpfE --timeout=600 --durations=0 \
-              --config-files=build_tools/pkgci/external_test_suite/pytorch_cpu_llvm_task.json
+              --config-files=SHARK-TestSuite/iree_tests/configs/config_pytorch_models_cpu_llvm_task.json
+      - name: "Running SDXL pipeline benchmark"
+        run: |
+          source ${VENV_DIR}/bin/activate
+          bash SHARK-TestSuite/iree_tests/benchmarks/benchmark_sdxl_cpu.sh
```

Comment:
May want to put benchmarking in a separate job so we can choose to run benchmarks less frequently than tests (e.g. only on postsubmit or when a PR author opts in). Looking at the logs at https://github.com/iree-org/iree/actions/runs/8760237017/job/24045123320?pr=17117, testing takes ~6 minutes and benchmarking takes ~5 minutes)

Also did this only run one iteration? Benchmarks should run at least a few iterations to ensure stable results:
```
----------------------------------------------------------------------------------------------------
Benchmark                                          Time             CPU   Iterations UserCounters...
----------------------------------------------------------------------------------------------------
BM_tokens_to_image/process_time/real_time     282651 ms      2175540 ms            1 items_per_second=3.53793m/s
```

(separately we'll want to ensure that benchmark actually makes sense... fp16 on CPU isn't expected to be fast)

---

Maybe we should start with
* tests on CPU
* tests on GPU
* benchmarks on GPU

then add CPU benchmarks later...?

Reply:
Sure sgtm. Removed the cpu benchmarks for now and addressed all the other comments as well. Should be good to go